models/poe.py

```diff
@@ -0,0 +1,43 @@
+import math
+
+
+def default_sigmoidal(stake, engagement):
+    return 2000 / (1 + math.exp(-0.000001 * (stake * engagement)**0.68)) - 1000
+
+
+def calculate_voting_weight(validator, params):
+    f = params['f']
+    self_stake = validator['self_stake']
+    delegation = validator['delegation']
+    engagement = validator['engagement']
+
+    return f(self_stake + delegation, engagement)
+
+
+def calculate_reward(validator, total_voting_weight, params):
+    inflation_rate = params['inflation_rate']
```

Comment:
Style nit:

Isn't `params['inflation_rate']` the same as `params.inflation_rate` in python? (It's been some years) If so, I prefer the second format here, as these are predefined keys, not just anything.

Reply:
Turns out my memory failed me and I spent too much time with TypeScript recently.

However, [dataclasses](https://docs.python.org/3/library/dataclasses.html) look nice. But that is just a thought for the future

Reply:
Yes, those look nice. Shall I open an issue?

Reply:
Sure. We can do it later for sure.

Reply:
https://github.com/confio/ProofOfEngagement/issues/4